semconv/internal/v2/http.go

```diff
@@ -254,7 +256,7 @@ func (c *HTTPConv) proto(proto string) attribute.KeyValue {
 		return c.HTTPFlavorKey.String("1.0")
 	case "HTTP/1.1":
 		return c.HTTPFlavorKey.String("1.1")
-	case "HTTP/2":
+	case "HTTP/2", "HTTP/2.0":
```

Comment:
I think that your issue comes from `net/http` NewRequestWithContext. It uses hardcoded proto value of `HTTP/1.1`.
https://cs.opensource.google/go/go/+/refs/tags/go1.20.1:src/net/http/request.go;l=892

This value gets filled in from the `otel-go` ClientRequest, and your script gets a span with `http.flavor` `1.1`.

My fix to this is re-fill in the `http.flavor` value from the `otel-go` ClientResponse. With this patch, you'll get a span with `http.flavor` `http/2.0` which is actually used.
https://cs.opensource.google/go/go/+/refs/tags/go1.20.1:src/net/http/h2_bundle.go;l=5998

As you said, although "http/2.0" is not along with the otel specification, I think it would be better to handle the input from the standard library too. So I added the case for `HTTP/2.0` to be converted into `2.0`. With this fix, you'll get a span with `http.flavor` `2.0`. That fix is the code line you commented on. That is not primary point of the issue.

I'm not sure should I add same kind of fix for http3, but one of the implementation of http3 generates a request with proto value `HTTP/3.0`, not `HTTP/3`. This would be converted into `http.flavor` `http/3.0`, not `3.0`.
https://github.com/quic-go/quic-go/blob/9488539a494de039d26b7761ae2af29bb7592b83/http3/request.go#L76